guides/telephony/twilio-daily-webrtc.mdx

```diff
@@ -1,197 +1,265 @@
-    parser.add_argument("-t", type=str, help="Token")
-    parser.add_argument("-i", type=str, help="Call ID")
-    parser.add_argument("-s", type=str, help="SIP URI")
-    config = parser.parse_args()
-
-    asyncio.run(main(config.u, config.t, config.i, config.d, config.s))
-
-async def main(room_url: str, token: str, callId: str, sipUri: str):
-    async with aiohttp.ClientSession() as session:
-        # diallin_settings are only needed if Daily's SIP URI is used
-        # If you are handling this via Twilio, Telnyx, set this to None
-        # and handle call-forwarding when on_dialin_ready fires.
-        diallin_settings = None
-
-        transport = DailyTransport(
-            room_url,
-            token,
-            "Chatbot",
-            DailyParams(
-                api_url=daily_api_path,
-                api_key=daily_api_key,
-                dialin_settings=diallin_settings,
-                audio_in_enabled=True,
-                audio_out_enabled=True,
-                camera_out_enabled=False,
-                vad_enabled=True,
-                vad_analyzer=SileroVADAnalyzer(),
-                transcription_enabled=True,
-            )
-        )
+resp = VoiceResponse()
+resp.play(url="https://your-hold-music.mp3", loop=10)
+return str(resp)
+```
+
+### ❌ Pitfall: Using `<Pause>` instead of hold music
 
-        # ... your bot code here
-
-        @transport.event_handler("on_dialin_ready")
-        async def on_dialin_ready(transport, cdata):
-            # For Twilio, Telnyx, etc. You need to update the state of the call
-            # and forward it to the sip_uri..
-            try:
-                # The TwiML is updated using Twilio's client library
-                call = twilioclient.calls(callId).update(
-                    twiml=f'<Response><Dial><Sip>{sipUri}</Sip></Dial></Response>'
-                )
-            except Exception as e:
-                raise Exception(detail=f"Failed to forward call: {str(e)}")
+Don't use the TwiML `<Pause>` element to wait for the bot to initialize:
+
+```python
+# DON'T DO THIS
+resp = VoiceResponse()
+resp.pause(length=10)  # This can cause connection issues
 ```
+
+Twilio can only pause a call for a short duration (~5 seconds), which may not be enough time for the Daily SIP setup to complete.
+
+### ✅ Best Practice: Handle multiple `on_dialin_ready` events
+
+If your Daily room has multiple SIP endpoints, use a flag to ensure you only forward the call once:
+
+```python
+call_already_forwarded = False
+
+@transport.event_handler("on_dialin_ready")
+async def on_dialin_ready(transport, cdata):
+    nonlocal call_already_forwarded
+
+    if call_already_forwarded:
+        logger.info("Call already forwarded, ignoring this event.")
+        return
+
+    # Forward the call...
+    call_already_forwarded = True
+```
+
+<Note>
+  A single SIP endpoint is sufficient for the initial connection. A second SIP
+  endpoint is needed only if you plan to forward the call.
```

Comment:
Only need extra endpoints if we want the bot to recover an additional sip participant. 

A dialout won't fire a dialin_ready